plugin-api/src/main/java/org/hyperledger/besu/plugin/services/securitymodule/SecurityModule.java

```diff
@@ -21,11 +25,12 @@
  * interface ensures cryptographic functions required by Ethereum are available to the application
  * at large, without releasing the content of the private key.
  */
+@Unstable
 public interface SecurityModule {
 
   /**
    * @param dataHash The Keccack hash of a set of data, which is to be signed.
-   * @return the signature (R, S, recId) generated by signing the hash with the node key
+   * @return the signature (R, S) generated by signing the hash with the node key
    */
   Signature sign(Bytes32 dataHash);
```

Comment:
Since the crypto provider can fail signing, calculating key agreement etc. I think we should have these methods throwing a custom exception, perhaps a SecurityModuleException.

Reply:
👍